src/Engines/QueryBuilderEngine.php

```diff
@@ -99,8 +99,10 @@ public function count()
      */
     public function filtering()
     {
+        $eagerLoads = array_keys($this->query->getEagerLoads());
```

Comment:
Or maybe we can extract this to own method that handles the exception when using query builder.

```php
    $eagerLoads = $this->getEagerLoads();
----------
    /**
     * Get eager loads key if eloquent.
     *
     * @return array
     */
    private function getEagerLoads()
    {
        if ($this->query_type == 'eloquent') {
            return array_keys($this->query->getEagerLoads());
        }

        return [];
    }
```